core/state_transition.go

```diff
@@ -213,7 +213,7 @@ func TransactionToMessage(tx *types.Transaction, s types.Signer, baseFee *big.In
 	}
 	// If baseFee provided, set gasPrice to effectiveGasPrice.
 	if baseFee != nil {
-		if msg.FeeCurrency != nil {
+		if tx.Type() == types.CeloDynamicFeeTxType {
```

Comment:
Is this change safe for the case where `tx.Type() == types.CeloDynamicFeeTxType` but `msg.FeeCurrency == nil`? Unfortunately, we didn't forbid `nil` currencies for CIP-64.

Reply:
Yes, the `ConvertGoldToCurrency` function just returns the passed in value in that case (here `baseFee`, which has the same effect as omitting the if case).